draft-private-access-tokens.md

```diff
@@ -416,15 +440,91 @@ Issuance assumes the Client has the following information, derived from a given
   corresponding to the Origin identified by TokenChallenge.origin_name.
 - Issuer name public key (ISSUER_NAME_KEY), a public key used to encrypt requests
   corresponding to the Issuer identified by TokenChallenge.issuer_name.
+  
+Clients also need to be able to generate an ANON_ORIGIN_ID value that corresponds
+to the ORIGIN_NAME, to send in requests to the Mediator.
+
+ANON_ORIGIN_ID MUST be a stable and unpredictable 32-byte value computed by the Client.
+Clients MUST NOT change this value across token requests for the same ORIGIN_NAME. Doing
+so will result in token issuance failure.
 
-Issuance also assumes that Issuers maintain local state for each distinct pair of Client
-and Origin. In particular, for each pair, Issuers maintain a stable mapping from ANON_CLIENT_ID
-to ORIGIN_NAME and ANON_ORIGIN_ID_PRIME values, as well as policy state about tokens
-that have already been issued to a Client. Policy state is custom to each implementation:
-it could include a simple counter to track the number of tokens issued to a
-ANON_CLIENT_ID, or have more detailed timestamp information.
+One possible mechanism for implementing this identifier is for the Client to store a mapping
+between the ORIGIN_NAME and a randomly generated ANON_ORIGIN_ID for future requests. Alternatively,
+the Client can compute a PRF keyed by a per-client secret (CLIENT_SECRET) over the ORIGIN_NAME,
+e.g., ANON_ORIGIN_ID = HKDF(secret=CLIENT_SECRET, salt="", info=ORIGIN_NAME).
 
-Finally, Issuance assumes that the Client and Mediator have a secure and
+### Mediator State {#mediator-state}
+
+Issuance assumes that Mediators maintain local state for each Client. The mechanism
+of identifying a Client is specific to each Mediator, and is not defined in this document.
+As examples, the Mediator could use device-specific certificates or account authentication
+to identify a Client.
+
+Mediators are expected to know the ISSUER_POLICY_WINDOW for any ISSUER_NAME to which
+they allow access. This information can be retrieved using the URIs defined in {{setup}}.
+
+For each pair of Client and Issuer, the Mediator maintains a specific policy window
+start and end time for each Issuer requested by the Client. Associated with this window,
+the Mediator determines a CLIENT_GROUP_ID value for the Client-Issuer pair, which
+changes every time the policy window elapses. The Mediator additionally maintains
+a counter of successful tokens issued within the policy window for each ANON_ORIGIN_ID
+requested by the Client, as well as a GROUP_ORIGIN_ID that it received for each
+ANON_ORIGIN_ID.
+
+The Mediator assigns CLIENT_GROUP_ID values to Clients in way that segments them into
+groups of roughly equal size. For example, at any time, Clients could be spread
```

Comment:
Say something about the group size -- that the Mediator chooses them, and the tradeoff is that an overly large group size makes it possible for a Mediator to learn about the Origin, and that an overly small group size makes it possible for the Issuer to learn about the Client. A Mediator SHOULD start by spreading users among as many groups as it can. With a large enough number of Clients, a Mediator SHOULD use at least ten groups at any given time, with each group having no fewer than thousands of Clients in it.

Reply:
The incentives for choosing a group size are a bit odd, that it seems like the mediator should always want to minimzize it, while the issuer wants to maximize it. I guess issuers will have minimum requirements for mediators that partner with them and vice versa.

Reply:
It's not clear to me how a large group reveals information about the origin. @janaiyengar, can you elaborate?

Reply:
@dvorak42 in order to learn info, I think it would actually be mediators preferring fewer larger groups (to learn about origins) and issuers preferring more smaller groups (to track clients)

Reply:
@janaiyengar I've taken your text and added it in here.